framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -717,6 +717,10 @@ assert getTopAnnotation(a1) == getTopAnnotation(a2) :
         }
         return newglbs;
     }
+    
+    protected boolean annotationMirrorsAreEqual(AnnotationMirror a1, AnnotationMirror a2) {
```

Comment:
I was planning to get rid of this method today and override the public method as you say (the other extension points I added were easier to justify, but this one isn't), so that's why I directed your attention away from the framework code. But of course you would still see a call to this from the Lock Checker code. Sorry about that.